### ISSUE TITLE
using system package as dependency in cmake_external 

### ISSUE THREAD

**Issue by bbarnes52** [bazelbuild/rules_foreign_cc#75]
I'd like to depend on a system package. When invoking cmake directly, I would just set CMAKE_PREFIX_PATH to the appropriate path. Is there a mechanism to accomplish the same with `cmake_external`?

The error I get is 
```
Could not find the required component 'foo'. The following CMake error indicates that you either need to install the package with the same name or change your environment so that it can be found.
```

I imagine the solution is to copy the package into $EXT_BUILD_DEPS, or else add the package path to $CMAKE_PREFIX_PATH. However, I have been unable to figure out how to accomplish this with the cmake_external API.

Thanks as always for your help, and let me know if there's a forum you would prefer I use for such questions!

**Comment by bbarnes52** [comment]
If I pass `cmake_options=-DCMAKE_PREFIX_PATH` = /path/to/package, it works as expected, but I assume that's only because this particular rule has no dependencies on other bazel rules.

**Comment by irengrig** [comment]
Yep, you are right.

If you need to provide additional path for CMake to search for libraries, you should use CMAKE_PREFIX_PATH.
In #76 (+ #77 small correction) I made it so that the value, passed by the user for that property is merged with the value cmake_external is passing for dependencies in $EXT_BUILD_DEPS to be found.

Please note that the CMAKE_PREFIX_PATH should be passed in cache_entries dictionary property:
(All -Dkey=value CMake options should be passed there; I made it this way so that the merging of values worked better)

`cmake_external(
  name = ...,
  cache_entries = {
    "CMAKE_PREFIX_PATH": "/usr/local/abc;/usr/bin/...",
  }
)`

I also modified one of the tests for the generated script to check that the values are actually merged.
Hope that helps.

**Comment by bbarnes52** [comment]
Thank you!!

**Comment by bbarnes52** [comment]
After bumping to the latest commit and updating my BUILD to use cache_entries I get the following error:

```
Traceback (most recent call last):
        File "/home/brian/git/cloud/ros/tutorial/catkin_ws/src/hello_world/BUILD", line 11
                cmake_external(name = 'hello')
        File "/home/brian/.cache/bazel/_bazel_brian/19fc196dfacab0a373352db9ae53e0f1/external/rules_foreign_cc/tools/build_defs/cmake.bzl", line 31, in _cmake_external
                create_cmake_script(ctx.workspace_name, ctx.attr._targ...], <8 more arguments>)
        File "/home/brian/.cache/bazel/_bazel_brian/19fc196dfacab0a373352db9ae53e0f1/external/rules_foreign_cc/tools/build_defs/cmake_script.bzl", line 31, in create_cmake_script
                _merge_prefix_path(user_cache)
        File "/home/brian/.cache/bazel/_bazel_brian/19fc196dfacab0a373352db9ae53e0f1/external/rules_foreign_cc/tools/build_defs/cmake_script.bzl", line 62, in _merge_prefix_path
                user_cache.pop("CMAKE_PREFIX_PATH")
trying to mutate a frozen object
```

**Comment by irengrig** [comment]
Yep, sorry for that. Fix is in #79
